### DISCUSSION TITLE
How will Codespaces billing work with public and open-source orgs/repos?

### DISCUSSION THREAD

**Discussion by Nezteb** [community/community#15645] [Enterprise] [answered]
Currently GitHub will monetize codespaces in orgs that pay for GitHub Team or GitHub Enterprise Cloud.

What about purely open-source orgs? What if there is an awesome OSS project that I want to contribute to (via a GitHub codespace) that doesn't pay for those plans? If the repo's org/owner isn't a company but instead is just a org for organizing related repos, is there no way to (optionally) push the cost onto the user instead of the org?

As a lover of open-source, I'd like to be able to pay for the ability to create a codespace in _any_ repo I want, even if the repo's org/owner doesn't pay for one of the above plans.

**Comment by craiglpeters** [comment, marked as answer]
I am so excited that you are excited to use Codespaces for your open source project! Very soon we will be sharing the news that we will be 1) making Codespaces available to every developer, and 2) provide a monthly free usage amount automatically for every user (with the ability to go over with a payment method).

**Reply by eldadfilmx** [reply]
Good evening sir

**Reply by Nezteb** [reply]
How will that work for organizations who have popular open-source projects that they'd like demoable/usable from Codespaces?

The Codespaces usage is billed to the organization, not individual users.

**Reply by craiglpeters** [reply]
So glad you asked! Codespaces usage (and thus billing) will follow the user *unless* an org has opted into paying for usage (with granular control).
